lib/ex_plasma/output.ex

```diff
@@ -0,0 +1,197 @@
+  ...> }
+  iex> ExPlasma.Output.encode(output, as: :input)
+  <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 59, 154, 202, 0>>
+  """
+  def encode(%{} = output, as: :input), do: to_rlp_id(output)
+  def encode(%{} = output), do: output |> to_rlp() |> ExRLP.encode()
+
+  @doc """
+
+  ## Example
+
+  iex> encoded_position = << 59, 154, 202, 0>>
+  iex> ExPlasma.Output.decode_id(encoded_position)
+  %ExPlasma.Output{
+    output_data: nil,
+    output_id: %{
+      blknum: 1,
+      oindex: 0,
+      position: 1000000000,
+      txindex: 0
+    },
+    output_type: nil
+  }
+  """
+  def decode_id(data), do: data |> :binary.decode_unsigned(:big) |> do_decode()
+
+  @doc """
+  Encode an Output into RLP bytes
+
+  ## Example
+
+  iex> output = %{
+  ...>      output_id: nil,
+  ...>      output_type: 1,
+  ...>      output_data: %{output_guard: <<1::160>>, token: <<0::160>>, amount: 1}
+  ...>    }
+  iex> ExPlasma.Output.to_rlp(output)
+  [<<1>>, [<<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1>>, <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, <<1>>]]
+  """
+  @spec to_rlp(t()) :: binary()
+  def to_rlp(%{output_type: nil}), do: nil
+  def to_rlp(%{output_type: type} = output), do: output |> get_output_type(type).to_rlp()
+
+  @doc """
+  Encodes an Output identifer into RLP bytes. This is to generate
+  the `inputs` in a Transaction.
+
+  ## Example
+  """
+  @spec to_rlp_id(t()) :: binary()
+  def to_rlp_id(%{output_id: nil}), do: nil
+  def to_rlp_id(%{output_id: id}), do: ExPlasma.Output.Position.to_rlp(id)
+
+  @doc """
+  Validates the Output
+
+  ## Example
+
+  # Validate a Payment v1 Output
+
+  iex> encoded = <<245, 1, 243, 148, 205, 193, 229, 59, 220, 116, 187, 245, 181, 247, 21, 214,
+  ...>  50, 125, 202, 87, 133, 226, 40, 180, 148, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  ...>  0, 0, 0, 0, 0, 0, 0, 0, 136, 13, 224, 182, 179, 167, 100, 0, 0>>
+  iex> {:ok, output} = encoded |> ExPlasma.Output.decode() |> ExPlasma.Output.validate()
+
+  # Validate a Output position
+
+  iex> encoded_position = <<59, 154, 202, 0>>
+  iex> {:ok, id} = encoded_position |> ExPlasma.Output.decode_id() |> ExPlasma.Output.validate()
+  """
+  @spec validate(t()) :: validation_responses()
+  def validate(%{} = output) do
+    with {:ok, _data} <- do_validate_data(output),
+         {:ok, _id} <- do_validate_id(output) do
+      {:ok, output}
+    end
+  end
+
+  # Validate the output ID. Bypass the validation if it doesn't
+  # exist in the output body.
+  defp do_validate_id(%{output_id: nil} = output), do: {:ok, output}
+
+  defp do_validate_id(%{output_id: %{} = output_id}),
+    do: ExPlasma.Output.Position.validate(output_id)
+
+  # Validate the output type and data. Bypass the validation if it doesn't
+  # exist in the output body.
+  defp do_validate_data(%{output_type: nil} = output), do: {:ok, output}
+
+  defp do_validate_data(%{output_type: type} = output) when is_integer(type),
```

Comment:
seems like these are multiline functions and a `do ... end` would suite them better